### ISSUE TITLE
Improve media picker (Swipe gesture)

### ISSUE THREAD

**Issue by Tienisto** [localsend/localsend#376]
Currently, we use the media picker from [wechat_assets_picker](https://pub.dev/packages/wechat_assets_picker) which required little effort to implement (because we just use others implementation).

However, the picker seems to be overcomplicated and the selection hitbox is too small. A media picker that works like in iOS would be desirable where you can use the swipe gesture to select multiple images at once.

**Comment by electr1fy0** [comment]
Also, please just use Android's own photo picker for the Android app.

**Comment by Tienisto** [comment]
It seems that [image_picker](https://pub.dev/packages/image_picker) allows us to use the native media pickers. Howevery, it can only pick one video at a time and also: images and videos are treated separately.

There is a more "beautiful" package called [gallery_picker](https://pub.dev/packages/gallery_picker) that has a more neat UI.

**Comment by Tienisto** [comment]
I have implemented the native media picker on Android and iOS but there is still a bug that the original file name gets lost. As soon as the flutter team fixes this, the native image pickers can be used.

Flutter bug:
https://github.com/flutter/flutter/issues/54815

Branch with native picker:
https://github.com/localsend/localsend/tree/use-native-media-picker

**Comment by mohitbhatia96** [comment]
Just discovered Local Send, swipe to selct multiple images on iPhone woud be very good. That is the only thing stopping me from using this app daily.

**Comment by Tienisto** [comment]
I just checked [image_picker](https://pub.dev/packages/image_picker) again, and they still didn't fix the issue that file names are lost: https://github.com/flutter/flutter/issues/64685

Retaining file names is too important to drop for a better UX.

**Comment by nbarari** [comment]
Is the flutter issue limited to Android? If so, is there any possibility of iOS utilizing image_picker?

**Comment by Tienisto** [comment]
This problem also exists on iOS (this issue is mentioned somewhere)

**Comment by KatyLB** [comment]
As a stop-gap, is it possible to have a setting you can enable or disable that when you click on an item it selects it instead of previewing it? I don't know if this sort of override is possible. Agree that the small checkbox and selection of one at a time is annoying when transferring hundreds of files.

**Comment by ilqvya** [comment]
Omg, I wanted to transfer all photos I have wihch is 5000+, are you really think I'm gling to pick every one manually ? lmao

**Comment by iSevenDays** [comment]
+1
I need to send a couple hundred photos. Checking them manually is not feasible

**Comment by dginovker** [comment]
+1, my girlfriend doesn't use localsend because multiselect sucks

**Comment by Tienisto** [comment]
This is a limitation by the Flutter ecosystem. Feel free to propose a Flutter package that supports swipe gesture and retains the file name.

The [image_picker](https://pub.dev/packages/image_picker) is near perfect, but fails to retain the file name: https://github.com/flutter/flutter/issues/54815

The current [wechat_assets_picker](https://pub.dev/packages/wechat_assets_picker) retains the file name but has no swipe gesture. They have an open issue https://github.com/fluttercandies/flutter_wechat_assets_picker/issues/175 that is exactly what we want.

**Comment by firecannons** [comment]
Yes, this would a nice feature, I am another person who would like to drag select 100+ media assets at a time. There is recent work happening on the wechat flutter asset picker about this topic. I have been working on this for a few weeks,  https://github.com/fluttercandies/flutter_wechat_assets_picker/pull/658, but https://github.com/fluttercandies/flutter_wechat_assets_picker/pull/660 will likely be the change that is accepted.

I tried using image_picker, but found it did not work well to select many assets. The image_picker package uses the operating system provided picker, and it becomes much slower as you select more assets. For example, in my testing, when 100+ images were selected, it took minutes to finish if it did at all. Maybe the native, operating system provided photo picker does a file copy operation before it returns. The wechat picker does not does not copy the selected media files, so it is much faster.

There are some other image pickers available in flutter, for example [this one](https://pub.dev/packages/insta_assets_picker), and others, but they don't seem to support drag select, and some are a bit dated now.

I would love to see this in LocalSend and make it even more awesome. It would be such a nice feature.

**Comment by ssJvirtually** [comment]
Loved the app getting 40MB/s transfer speeds from iPhone13 to PC

The only drawback is   multi select or select all options


Hope that gets fixed soon

**Comment by JustAdreamerFL** [comment]
For the time being, just use file picking and pick those images that way. This works on Android, idk if on ios too.

**Comment by ssJvirtually** [comment]
In ios it's not that straight forward 
you have to copy the photos to files app , which uses more storage unnecessarily

**Comment by JustAdreamerFL** [comment]
Bruh

**Comment by Tienisto** [comment]
Thanks to the effort of flutter_wechat_assets_picker, the latest version (1.17.0) supports dragging/swiping multiple media files!
